twisted/test/test_compat.py

```diff
@@ -315,6 +317,26 @@ def test_python3(self):
 
 
 
+class PYPYTest(unittest.SynchronousTestCase):
+    """
+    Identification of PyPy.
+    """
+
+    def test_pypy(self):
+        """
+        On PyPy, C{_PYPY} is True.
+        """
+        if 'PyPy' in sys.version:
+            self.assertTrue(_PYPY)
+
```

Comment:
That's not too convincing, since if the test fails (and it rarely will), the assert will give a stack trace on the exact line, and it is not too hard to figure out.
As for why PY3Tests chose the approach it did, I won't comment on that.  That can be cleaned up in a separate ticket.